youtube_dl/extractor/weibo.py

```diff
@@ -0,0 +1,153 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import json
+import random as rnd
+import re
+
+from ..compat import (
+    compat_urllib_parse_urlencode as urlencode,
+    compat_urllib_request as request,
+    compat_urlparse as parse,
+)
+from ..utils import (
+    js_to_json,
+)
+
+
+class WeiboIE(InfoExtractor):
+    _VALID_URL = r'https?://weibo\.com/[0-9]+/(?P<id>[a-zA-Z0-9]+)'
+    _TEST = {
+        'url': 'https://weibo.com/6275294458/Fp6RGfbff?type=comment',
+        'info_dict': {
+            'id': 'Fp6RGfbff',
+            'ext': 'mp4',
+            'title': 'You should have servants to massage you,... 来自Hosico_猫 - 微博',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        headers = {
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept-Language': 'en,zh-CN;q=0.9,zh;q=0.8',
+            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.84 Safari/537.36',
+            'Upgrade-Insecure-Requests': '1',
+        }
+        # to get Referer url for genvisitor
+        webpage, urlh = self._download_webpage_handle(url, video_id, headers=headers, note="first visit the page")
+
+        visitor_url = urlh.geturl()
+
+        data = urlencode({
+            "cb": "gen_callback",
+            "fp": '{"os":"2","browser":"Gecko57,0,0,0","fonts":"undefined","screenInfo":"1440*900*24","plugins":""}',
+        }).encode()
+        headers = {
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept': '*/*',
+            'Referer': visitor_url,
+        }
+
+        r_genvisitor = request.Request(
+            'https://passport.weibo.com/visitor/genvisitor',
+            data=data,
+            headers=headers,
+        )
+        webpage, urlh = self._download_webpage_handle(r_genvisitor, video_id, note="gen visitor")
+
+        p = webpage.split("&&")[1]  # split "gen_callback && gen_callback(...)"
+        i1 = p.find('{')
+        i2 = p.rfind('}')
+        j = p[i1:i2 + 1]  # get JSON object
+        d = json.loads(j)
+        tid = d["data"]["tid"]
+        cnfd = "%03d" % d["data"]["confidence"]
+
+        param = urlencode({
+            'a': 'incarnate',
+            't': tid,
+            'w': 2,
+            'c': cnfd,
+            'cb': 'cross_domain',
+            'from': 'weibo',
+            '_rand': rnd.random()
+        })
+        gencallback_url = "https://passport.weibo.com/visitor/visitor?" + param
+        webpage, urlh = self._download_webpage_handle(gencallback_url, video_id, note="gen callback")
```

Comment:
You can drop the ```webpage, urlh = ``` part if they are not used. And it's better to have a comment for why this request is necessary